Add colab implementation of WitWidget

jameswex opened the pull request:
- Motivation for features / changes

Adds colab implementation of WitWidget to enable use of What-If Tool directly in colab notebooks. This implementation differs by necessity from the Jupyter notebook widget.

- Technical description of changes

Added Colab implemenation of WitWidget, which uses the colab-recommended way of passing information from python to JS and passing information from JS to python.

Added example colab notebook showing its use.

- Screenshots of UI changes

![wit-colab](https://user-images.githubusercontent.com/15835086/50703216-6a9ab600-1021-11e9-8c24-2eab2fc9ab0b.png)

- Detailed steps to verify changes work correctly (as executed by you)

Run the colab notebook that is part of the PR and verify all What-If Tool features work correctly.

- Alternate designs / implementations considered

N/A

jameswex commented:
@tolga-b Please take a look, thanks!

tolga-b commented:
Thanks James, it looks great. You may want to add a direct link to colab in the readme as we discussed: https://colab.sandbox.google.com/github/jameswex/tensorboard/blob/colab/tensorboard/plugins/interactive_inference/WIT_in_colab.ipynb. Format is github/[repo-link]/blob/[branch-name]/[path-within-repo-to-file].

tolga-b commented on tensorboard/plugins/interactive_inference/witwidget/notebook/colab/wit.py:
```diff
@@ -12,4 +12,319 @@
+def infer_examples():
+  WitWidget.widget.infer()
+output.register_callback('notebook.InferExamples', infer_examples)
+
```
I think its suggested to have two spaces between module functions.

jameswex replied:
done

jameswex commented:
thanks tolga, added that direct link in the spot at the bottom and also another spot near the top of the readme where i mention colab. also updated widget to be able to handle multiple witwidgets in a single colab and tested it with two witwidgets.

stephanwlee commented on tensorboard/plugins/interactive_inference/witwidget/notebook/colab/wit.py:
```diff
@@ -12,4 +12,339 @@
+
+
+# HTML/javascript for the WIT frontend.
+WIT_HTML = """
```
It would be wonderful if this is created/went through compilation/Vulcanization and read as a string. Would it be possible?
